src/calliope/postprocess/functions.py

```diff
@@ -0,0 +1,113 @@
+"""Built-in postprocessing functions."""
+
+import xarray as xr
+
+from calliope.postprocess.orchestrator import PostprocessContext, postprocessor
+
+
+def _get_data(name: str, context: PostprocessContext) -> xr.DataArray:
+    """Get an element from either inputs or results."""
+    if name in context.results.keys():
+        data = context.results[name]
+    else:
+        data = context.inputs[name]
+    return data
```

Comment:
```suggestion
def _get_data(name: str, context: PostprocessContext) -> xr.DataArray:
    """Get an element from either inputs or results."""
    data = context.results.get(name, context.inputs[name])
    return data
```

Reply:
No, it shouldn't. It will fall back on `context.inputs[name]` if it isn't found in results. One thing we should possibly do is return the default value defined in math if it isn't found in either of those, rather than failing with a key error. There's nothing worse than your model failing in postprocessing after a long solve!